### PR TITLE
multi: remove experimential build flag for assumechanvalid

### LINKED ISSUE #4581: Remove requirement for experimental flag for --routing.assumechanvalid
### Background
lncli sendpayment fails with error even lncli channelbalance responds with 

lncli channelbalance
{
    "balance": "1990950",
    "pending_open_balance": "0"
}

channel has 3 confirmations

### Your environment

v0.11.0-beta
4.15.0-112-generic #113-Ubuntu x86_64 x86_64 x86_64 GNU/Linux
neutrino backend

### PR DESCRIPTION
**Pull request by carlaKC**
This PR removes the `experimental` build flag for the `assumechanvalid` config option.

Fixes #4581, my understanding from the issue title/comments is that we want to remove this flag entirely. 

Alternatively, we could keep the `experimental` flag but update our release script to [include it](https://github.com/carlaKC/lnd/tree/4581-includeexperimentalrelease) so that we release binaries that include the `assumechanvalid` option, also suggested in the [issue](https://github.com/lightningnetwork/lnd/issues/4581#issuecomment-685054891).

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -269,12 +269,12 @@ vendor:
...
 	@$(call print, "Building iOS framework ($(IOS_BUILD)).")
 	mkdir -p $(IOS_BUILD_DIR)
-	$(GOMOBILE_BIN) bind -target=ios -tags="mobile $(DEV_TAGS) autopilotrpc experimental" $(LDFLAGS) -v -o $(IOS_BUILD) $(MOBILE_PKG)
+	$(GOMOBILE_BIN) bind -target=ios -tags="mobile $(DEV_TAGS) autopilotrpc" $(LDFLAGS) -v -o $(IOS_BUILD) $(MOBILE_PKG)
 
 android: vendor mobile-rpc
 	@$(call print, "Building Android library ($(ANDROID_BUILD)).")
 	mkdir -p $(ANDROID_BUILD_DIR)
-	$(GOMOBILE_BIN) bind -target=android -tags="mobile $(DEV_TAGS) autopilotrpc experimental" $(LDFLAGS) -v -o $(ANDROID_BUILD) $(MOBILE_PKG)
+	$(GOMOBILE_BIN) bind -target=android -tags="mobile $(DEV_TAGS) autopilotrpc" $(LDFLAGS) -v -o $(ANDROID_BUILD) $(MOBILE_PKG)
```
Comment (Roasbeef) [inline Makefile:277]: We should double check that there's nothing else the mobile builds need from the experimental flag as well.

**Review by Roasbeef — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (halseth) [reply Makefile:277]: I'm pretty sure this was the only experimental option left

File: sample-lnd.conf [hunk 2]
```diff
@@ -408,6 +408,9 @@ bitcoin.node=btcd
 ; Used to help identify ourselves to other bitcoin peers (default: 0.11.0-beta).
 ; neutrino.useragentversion=0.11.0-beta
 
+; Skip checking channel spentness during graph validation for neutrino.
```
Comment (halseth) [inline sample-lnd.conf:411]: Maybe add either here or the doc of the flag the downsides to using it.

**Review by halseth — CHANGES_REQUESTED** [review]
LGTM, just think we need to state more clearly the risk associated with using the flag.

File: Makefile (hunk 1 above)
Comment (carlaKC) [reply Makefile:277]: Checked `falafel` and it's not used there. Not sure if there's anything else that needs to be checked?

**Review by halseth — APPROVED** [review]
LGTM 🌮

**Review by Roasbeef — APPROVED** [review]
LGTM 🍟

**Pull request merged**